Added default command for LEDSubsystem

Jetblackdragon opened the pull request:
Added command that turns on a blue light when note is in the index and a green light when the note is aligned.

Jetblackdragon committed:
Added default command for LEDSubsystem

Added command that turns on a blue light when note is in the index and a green light when the note is aligned.

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java:
```diff
@@ -178,6 +180,18 @@ public void setAngleSpeed(double Speed) {
+				setLauncherSpeedEntry.getDouble(0),
+				launcherTopEncoder.getVelocity(),
+				FLYWHEEL_VELOCITY_TOLERANCE);
+		// if(MathUtil.isNear(setLauncherSpeedEntry.getDouble(0), launcherTopEncoder.getVelocity(), )){
+		// 	return true;
+		// }
+		// return false;
```
Please cleanup / remove commented code that is no longer used.

bobtseattle commented on src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java:
```diff
@@ -0,0 +1,42 @@
+
+	@Override
+	public boolean isFinished() {
+        LEDSubsystem.setRED_LED();
```
Looks to be more indentation issues.  These are often driven when using tabs vs spaces.

bobtseattle commented on src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java:
```diff
@@ -0,0 +1,42 @@
+            LEDSubsystem LEDSubsystem, 
+            IntakeSubsystem intakeSubsystem, 
+            LauncherSubsystem launcherSubsystem) {
+		this.LEDSubsystem = LEDSubsystem;
```
Make sure spotless was run to properly format code. Looks to be indentation issues in this block.

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java:
```diff
@@ -178,6 +180,18 @@ public void setAngleSpeed(double Speed) {
 
+	public boolean atTargetSpeed() {
+
+		return MathUtil.isNear(
```
With the ability to run independently, seems like it should be checking both motor speeds, especially for shooting into amp or just shot into amp and now shooting into speaker?
Is that the right tolerance, 0.1?

bobtseattle reviewed (changes requested):
Some cleanup, and making sure logic / tolerance is what we need it to be?

Jetblackdragon committed:
Logic errors + spotless

fixed some logic errors and did a spotless run

Jetblackdragon committed:
merge conflict

merge conflict

Jetblackdragon committed:
conflicts

conflicts

Jetblackdragon committed:
Merge branch 'main' into LED-

Jetblackdragon committed:
Removed duplicate method

Had a duplicate at speed method so removed the new one.

bobtseattle reviewed (dismissed):
LGTM.  We can add other color options in a different PR.

TAKBS2412 commented on src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java:
```diff
@@ -0,0 +1,42 @@
+
+	@Override
+	public boolean isFinished() {
+		ledSubsystem.setRED_LED();
```
Why are we setting the leds to red in `isFinished()`?

kirbt commented:
It might be useful to add an LED color for communicating to the drivers that the robot's intake is off

jamesdooley4 reviewed (changes requested):
LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.

iamawesomecat commented:
> LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.

Do we want to close it until then?

TAKBS2412 commented:
> > LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.
> 
> Do we want to close it until then?

I've converted it into a draft for now.

Jetblackdragon committed:
Intake checks and disable

Added color to check if intake is on or not and added a end command to disable the leds

Jetblackdragon committed:
Spotless

Spotless

Jetblackdragon committed:
Bugs

Fixed bugs

Jetblackdragon committed:
Merge branch 'main' into LED-

Jetblackdragon committed:
Spotless

spotless

Jetblackdragon committed:
Changed deleted motors

Fixed methods that used outdated motors

Jetblackdragon committed:
Added parameter for led command

Added parameter for led command

Jetblackdragon committed:
enabled led subsystem

enabled led subsystem

bobtseattle commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -241,6 +241,13 @@ public boolean getSensorOverride() {
 	}
 
+	public boolean isIntakeOn() {
+		return (intakeMotorFront.get() != 0
```
Just want to verify this logic is correct.  I may be wrong (I am assuming we are giving drive team feedback that intake can intake a note), but seems like if intake is in reject mode (assuming < 0), that we shouldn't be lighting up the intake on signal.

Jetblackdragon replied:
Drive team told me to leave rejecting to be yellow as well.

bobtseattle commented on src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java:
```diff
@@ -0,0 +1,50 @@
+
+	@Override
+	public void execute() {
+		if (launcherSubsystem.isAtSpeed()) { // Checks if launcher is ready
+			ledSubsystem.setGREEN_LED();
+		} else if (intakeSubsystem.indexSensorHasNote()) { // Checks if note is in feeder
+			ledSubsystem.setBLUE_LED();
+		} else if (intakeSubsystem.isIntakeOn()) { // Checks if intake is on
+			ledSubsystem.setYELLOW_LED();
+		} else { // Everything else including intake off
+			ledSubsystem.setRED_LED();
+		}
```
Is this logic correct?  Should it be green if it is up to speed but does not have a note?
Should there be a color difference between a note just entering index and a note in position to be launched?

Jetblackdragon replied:
To be green it has been updated to check if it has a note as well. New color violet has been added to check if the feeder sensor detects a note.

bobtseattle reviewed (commented):
It would be good to make sure there is ambiguity between the state of robot and LEDs.

Jetblackdragon committed:
Added color violet

New violet color to check if feeder sensor sees a note.

Jetblackdragon committed:
Merge branch 'robototes:main' into LED-

Jetblackdragon committed:
Changed isIntakeOn

Changed isIntakeOn method to only check for intaking and not also rejecting

Jetblackdragon committed:
Merge branch 'LED-' of https://github.com/Jetblackdragon/Crescendo2024 into LED-

Jetblackdragon committed:
Merge branch 'main' into LED-

bobtseattle reviewed (dismissed):
LGTM - We can deal with back intake during testing.  Though, should deal with failing formatting failures.

Jetblackdragon committed:
Merge branch 'robototes:main' into LED-

Jetblackdragon committed:
Update Controls.java

Jetblackdragon committed:
Merge Conflicts

Fixed merge conflicts

Jetblackdragon committed:
Merge branch 'main' into LED-

Jetblackdragon committed:
Merge branch 'robototes:main' into LED-

Jetblackdragon committed:
Testing changes

testing changes on led stuff

jamesdooley4 commented on src/main/java/frc/team2412/robot/Robot.java:
```diff
@@ -43,6 +43,9 @@ public static Robot getInstance() {
 	private static final boolean sysIdMode = false;
 
+	// Turns on an off auto logic
+	private static final boolean autoEnabled = false;
```
Likely should turn autonomous on by default

jamesdooley4 reviewed (changes requested):
Need to re-enable auto

Jetblackdragon committed:
Turned on auto logic
